### PR TITLE
fix URI for Collections, and make other assorted typo and wordsmithing changes extracted from the browseable/children PR

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** #223

This fix is also in the browseable PR, but just breaking it out to get it in before that massive one. 

**Proposed Changes:**

1. use the correct Collection conformance class URI in the Collection spec

**PR Checklist:**

- [X] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [X] This PR has **no** breaking changes.
- [X] This PR does not make any changes to the core spec in the `stac-spec` directory (these are included as a subtree and should be updated directly in [radiantearth/stac-spec](https://github.com/radiantearth/stac-spec))
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-api-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,9 @@ The *[ogcapi-features](ogcapi-features)* folder describes how a STAC API can ful
 Features](http://docs.opengeospatial.org/is/17-069r3/17-069r3.html) to expose individual `items` endpoints for search of
 each STAC collection. It also includes extensions that can be used to further enhance OAFeat.
 
+**STAC API - Collections:**
```
Comment (philvarner) [inline README.md:67]: this was missing, so adding.

File: collections/README.md [hunk 2]
```diff
@@ -7,16 +7,17 @@
   - [Example](#example)
 
 - **OpenAPI specification:** [openapi.yaml](openapi.yaml) ([rendered version](https://api.stacspec.org/v1.0.0-beta.5/collections))
-- **Conformance URI:** <http://stacspec.org/spec/api/1.0.0-beta.5/extensions/collections>
+- **Conformance URI:** <https://api.stacspec.org/v1.0.0-beta.5/collections>
```
Comment (philvarner) [inline collections/README.md:10]: this was copied wrong when Collections conf class was added, so fixing

File: overview.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # About
 
-The STAC API defines a RESTful JSON-based web API to browse and query [SpatioTemporal Asset Catalog](stac-spec/)
```
Comment (philvarner) [inline overview.md:3]: OGC API convention is to only use the term "web API" and avoid REST or RESTful

File: README.md [hunk 4]
```diff
@@ -12,9 +12,9 @@
 
 ## About
 
-The SpatioTemporal Asset Catalog (STAC) specification aims to standardize the way geospatial assets are exposed online and queried. 
+The SpatioTemporal Asset Catalog (STAC) family of specifications aim to standardize the way geospatial asset metadata is structured and queried.
 A 'spatiotemporal asset' is any file that represents information about the earth captured in a certain space and 
-time. The core STAC specification lives at [gitub.com/radiantearth/stac-spec](https://github.com/radiantearth/stac-spec).
+time. The core STAC specifications live in the GitHub repo [radiantearth/stac-spec](https://github.com/radiantearth/stac-spec).
```
Comment (m-mohr) [inline README.md:17]:
```suggestion
time. The core STAC specifications live in the GitHub repository [radiantearth/stac-spec](https://github.com/radiantearth/stac-spec).

```

File: README.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ Typically, a Feature is used when returning a single Item object, and FeatureCol
 JSON array of Item entities).
 
 The API can be implemented in compliance with the *[OGC API - Features](http://docs.opengeospatial.org/is/17-069r3/17-069r3.html)* standard 
-(we'll use OAFeat for shorthand). In this case STAC API can be thought of as a specialized Features API 
+(OAFeat as a shorthand). In this case STAC API can be thought of as a specialized Features API
```
Comment (m-mohr) [inline README.md:27]: Not sure, but this sounds weird to me due to the verb missing. Maybe make it easier to read for non-natives?
Or is it meant to be:
```suggestion
(OAFeat is a shorthand). In this case STAC API can be thought of as a specialized Features API 

```

File: collections/openapi.yaml [hunk 6]
```diff
@@ -77,20 +69,6 @@ paths:
                     rel: data
                     type: application/json
                     title: Information about the feature collections
-  '/conformance':
```
Comment (m-mohr) [inline collections/openapi.yaml:80]: Removing due to inheritance from Core, I guess?

**Review by m-mohr — APPROVED** [review]
Left some minor comments that should be adressed, but nothing that is really blocking.
Otherweise, looks good, thanks for the breakout PR.

File: collections/openapi.yaml (hunk 6 above)
Comment (philvarner) [reply collections/openapi.yaml:80]: Correct.

**Pull request merged**